src/main/java/ru/yandex/practicum/filmorate/storage/user/UserDbStorage.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    @Override
+    public List<Integer> addFriendship(int firstId, int secondId) {
+        checkUser(firstId);
+        checkUser(secondId);
+        final String sqlUpdateQuery = "UPDATE friends SET status = ? WHERE user_id = ? AND friend_id = ?";
+        final String sqlWriteQuery = "INSERT INTO friends (user_id, friend_id, status ) VALUES (?, ?, ?)";
+        final String checkQuery = "SELECT * FROM friends WHERE user_id = ? AND friend_id = ?";
+        SqlRowSet userRows = jdbcTemplate.queryForRowSet(checkQuery, firstId, secondId);
+        if (userRows.next()) {
+            jdbcTemplate.update(sqlUpdateQuery, true, firstId, secondId);
+            log.info("user с id = {} подтвердил дружбу с user id = {}", firstId, secondId);
+        } else {
+            jdbcTemplate.update(sqlWriteQuery, firstId, secondId, false);
+            log.info("user с id = {} подписался на user id = {}", firstId, secondId);
+        }
+        return List.of(firstId, secondId);
+    }
+
+    @Override
+    public List<Integer> removeFriendship(int firstId, int secondId) {
+        checkUser(firstId);
+        checkUser(secondId);
+        final String sqlDeleteQuery = "DELETE FROM friends WHERE user_id = ? AND friend_id = ?";
+        final String checkQuery = "SELECT status FROM friends WHERE user_id = ? AND friend_id = ?";
+        SqlRowSet userRows = jdbcTemplate.queryForRowSet(checkQuery, firstId, secondId);
+        SqlRowSet userRowsRevers = jdbcTemplate.queryForRowSet(checkQuery, secondId, firstId);
+        jdbcTemplate.update(sqlDeleteQuery, firstId, secondId);
+        log.info("user с id = {} убрал из друзей user id = {}", firstId, secondId);
+        return List.of(firstId, secondId);
+    }
+
+    @Override
+    public List<User> getFriendsListById(int id) {
+        checkUser(id);
+        final String sqlQuery = "SELECT id, email, login, name, birthday FROM users " +
+                "LEFT JOIN friends ON users.id = friends.friend_id " +
+                "WHERE user_id = ?";
+        log.info("запрос списка друзей user = {} отправлен", id);
+        return jdbcTemplate.query(sqlQuery, this::makeUser, id);
+    }
+
+    @Override
+    public List<User> getCommonFriendsList(int firstId, int secondId) {
+        checkUser(firstId);
+        checkUser(secondId);
+        final String sqlQuery = "SELECT id, email, login, name, birthday FROM friends " +
+                "LEFT JOIN users ON users.id = friends.friend_id " +
+                "WHERE friends.user_id = ? AND friends.friend_id IN " +
+                "( SELECT friend_id FROM friends " +
+                "LEFT JOIN users ON users.id = friends.friend_id " +
+                "WHERE friends.user_id = ? )";
+        log.info("запрос списка общих друзей user = {} и user = {} отправлен", firstId, secondId);
+        return jdbcTemplate.query(sqlQuery, this::makeUser, firstId, secondId);
+    }
+
+    private User makeUser(ResultSet rs, int rowNum) throws SQLException {
```

Comment:
Лучше не сокращать названия параметров
```suggestion
    private User makeUser(ResultSet resultSet, int rowNum) throws SQLException {
```